UI/TechTreeWnd.cpp

```diff
@@ -1583,20 +1604,108 @@ void TechTreeWnd::TechListBox::TechRow::Render() {
 }
 
 std::vector<GG::X> TechTreeWnd::TechListBox::TechRow::ColWidths(GG::X total_width) {
-    const GG::X GRAPHIC_WIDTH(ClientUI::Pts() * 2);
-    const GG::X NAME_WIDTH(ClientUI::Pts() * 18);
-    const GG::X COST_WIDTH(ClientUI::Pts() * 4);
-    const GG::X TIME_WIDTH(ClientUI::Pts() * 4);
-    const GG::X CATEGORY_WIDTH(ClientUI::Pts() * 8);
-
-    const GG::X DESC_WIDTH = std::max(GG::X1, total_width - GRAPHIC_WIDTH - NAME_WIDTH - COST_WIDTH - TIME_WIDTH - CATEGORY_WIDTH);
     std::vector<GG::X> retval;
-    retval.push_back(GRAPHIC_WIDTH);
-    retval.push_back(NAME_WIDTH);
-    retval.push_back(COST_WIDTH);
-    retval.push_back(TIME_WIDTH);
-    retval.push_back(CATEGORY_WIDTH);
-    retval.push_back(DESC_WIDTH);
+    // Early return for very small screen
+    if (total_width < (6 * 4 * ClientUI::Pts())) {
+        ErrorLogger() << "Not enough room for reasonable column space";
+        retval = {GG::X1, GG::X1, GG::X1, GG::X1, GG::X1, GG::X1};
+        return retval;
+    }
+
+    // Attempt to utilize user defined widths
+    // If not enough width to fit, attempt default widths
```

Comment:
Note, column headers do not scroll horizontally with listbox rows.
I've changed per request as I suspect this will not be a major issue until in-game adjustments are supported.